### PR TITLE
refactor: rewrite File implementation

### PR DESCRIPTION
**Pull request by lucacasonato**
This PR contains a rewritten `File` implementation that follows the spec at https://w3c.github.io/FileAPI/#file-section. This implementation passes all relevant File related web platform tests.

### CODE REVIEW

**Review by satyarohith — COMMENTED** [review]
Looks good.

File: op_crates/fetch/21_file.js [hunk 1]
```diff
@@ -287,8 +287,57 @@
     }
   }
 
-  window.__bootstrap.blob = {
+  const _name = Symbol("[[Name]]");
+  const _lastModfied = Symbol("[[LastModified]]");
```
Comment (kitsonk) [inline op_crates/fetch/21_file.js:291]: nitpic, alphabetise

File: op_crates/fetch/21_file.js [hunk 2]
```diff
@@ -287,8 +287,57 @@
...
+    [_name];
+    /** @type {number} */
+    [_lastModfied];
+
+    /**
+     * @param {BlobPart[]} fileBits 
+     * @param {string} fileName 
+     * @param {FilePropertyBag} [options] 
+     */
+    constructor(fileBits, fileName, options) {
```
Comment (kitsonk) [inline op_crates/fetch/21_file.js:304]:
```suggestion
    constructor(fileBits, fileName, options = {}) {
```

Would obviate the need for the optional chaining below and should still be spec compliant.

**Review by kitsonk — APPROVED** [review]
LGTM, but a couple minor nit picks.

File: op_crates/fetch/21_file.js (hunk 2 above)
Comment (lucacasonato) [reply op_crates/fetch/21_file.js:304]: Not really. When passing `null` as the third parameter, the `= {}` wouldn't catch, and thus options could still not be a WebIDL dictionary. The optional chaining prevents this.

**Pull request merged**